identity/src/keypair.rs

```diff
@@ -484,16 +664,72 @@ mod tests {
+        let pub_key = PublicKey::try_decode_protobuf_encoding(&hex_literal::hex!("080012a60430820222300d06092a864886f70d01010105000382020f003082020a02820201009c897f33e0d0b3297f2fe404ea5b7a98096b329693292aefc2d05ef1e82fd0e121ce74ec77d75ef4b532fa34dee2a19626f3389c6d2bb9b8de614e138302bc4254727a7ee35f7827f1094403bc2fe8e1f64d0e8a2a77e8f3a879f69f94a71f3589de184f5910d6b5270f58e684f71ddd3a3f486a4cb2c390194ee6e9b65f9f1dff7b8f6c0bf4e0c4ac683bd4ba2d2fd022fdaaa3db75e90e16662fc4b3aca4c9aa65514d51690cd372c2b96c61a1ed4f9298ec213d5398aa9120379477118391104deb77ab157a59b70714e95caa9b55d15fa386b0c80f36e50d738bdd10e0baa3c3eafb4703dec3d6a757601f18541eb87ae9111f60eae17d843cf1047dbf5a8982ad9ef0aa88f59b17689f1210a305f7da8a012c1a58e4e82b48811618e98cef13c9eb28ce6fcc589ea5d902149ee4f49f8b39758b349ca90be5a8bddf4a46bacaaa48aec1c0c6e996ab13f2cb351c351d40b0a7b8e0c12b366a8555c392b0aadf71fe746eb4f8ea0b829da6ddcc39081abdd40ea2f3d8778b9a3f06a480ef34234975e919c0d64d818f2e904a9f251c8669dbb1666cb2c28e955446fc7efd460d4677ed922ccff1e24bb5a8699e050075c7897a64daa1bc2f05e4132e76c4f72baea5d073042254236c116ea3e40540bb7986468b4468aadfadad068331ef9dbe13e4012196e8eb9f8cdba096c35f09e80893ea68f3253dc41053983855e50203010001")).unwrap();
+
+        roundtrip_protobuf_encoding(&priv_key, &pub_key);
+    }
+
+    fn roundtrip_protobuf_encoding(private_key: &Keypair, public_key: &PublicKey) {
+        assert_eq!(&private_key.public(), public_key);
+
+        let encoded_priv = private_key.encode_protobuf_encoding();
+        let decoded_priv = Keypair::try_decode_protobuf_encoding(&encoded_priv).unwrap();
+
+        assert_eq!(
+            private_key.public().to_peer_id(),
+            decoded_priv.public().to_peer_id(),
+            "PeerId from roundtripped private key should be the same"
+        );
+
+        let encoded_public = private_key.public().encode_protobuf_encoding();
+        let decoded_public = PublicKey::try_decode_protobuf_encoding(&encoded_public).unwrap();
+
+        assert_eq!(
+            private_key.public().to_peer_id(),
+            decoded_public.to_peer_id(),
+            "PeerId from roundtripped public key should be the same"
+        );
     }
```

Comment:
As a heads-up: I added these tests before I realised that our protobuf encoding function is not actually correct yet (see other comments). You will need to adapt these.